core/src/main/java/org/mule/runtime/core/api/transaction/TransactionFactory.java

```diff
@@ -24,6 +27,19 @@
    */
   Transaction beginTransaction(MuleContext muleContext) throws TransactionException;
 
+  /**
+   * Create and begins a new transaction
+   *
+   * @return a new Transaction
+   * @throws TransactionException if the transaction cannot be created or begun
+   * @param applicationName will be part of the notification
+   * @param notificationFirer allows the Mule container to fire notifications
+   */
+  Transaction beginTransaction(String applicationName, NotificationDispatcher notificationFirer,
```

Comment:
add a default implementation that delegates to the old `beginTransaction`. You may hackily obtain the muleContext from the `NotificationDispatcher`